### PR TITLE
feat: article about tolgee and next 13

### PR DESCRIPTION
**Pull request by stepan662**

### CODE REVIEW

File: blog/2023-08-01-tolgee-with-next-13.md [hunk 1]
```diff
@@ -0,0 +1,347 @@
+---
+slug: tolgee-with-nextjs-app-router
+title: 'Tolgee with Next.js app router'
```
Comment (JanCizmar) [inline blog/2023-08-01-tolgee-with-next-13.md:3]: This is exactly what no-one would search for. People are interested into "Translating Next.js 13 app", "The simplest way to localize Next.js app" etc. I don't even get what's this whole article about from the title.

File: blog/2023-08-01-tolgee-with-next-13.md [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+authors: [sgranat]
+description: 'Setting up Tolgee with Next.js new app router and server components.'
+tags: [next.js, server components, react]
+---
+
+With the introduction of Next.js 13 and the app router featuring React server components, there has been a strong push to enable Tolgee to work within this new paradigm.
+
+Until now, Tolgee was designed to function with regular React components. However, we now face the challenge of making in-context translation work with server components. Let's delve into the details.
```
Comment (JanCizmar) [inline blog/2023-08-01-tolgee-with-next-13.md:11]: There is no intro to the server components, so I don't know how it works.

File: blog/2023-08-01-tolgee-with-next-13.md [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+<!--truncate-->
+
+> #### How Tolgee In-Context Works
+>
+> Tolgee's unique method of enabling users to directly edit translations within the app is based on inserting invisible characters next to the actual translations. This creates a kind of `watermark` for each translation, making it detectable in the DOM and allowing precise localization.
+> This approach offers a non-intrusive way to empower users to translate in the context of their app.
+
+### The General Plan
+
+For server components, the approach involves including a complete key name within the invisible `watermark` characters. Additionally, the development translation files need to be loaded on the server to facilitate rendering. On the client side, the SDK will recognize the server-rendered `watermarks` and enable the in-context translation functionality.
```
Comment (JanCizmar) [inline blog/2023-08-01-tolgee-with-next-13.md:22]: Some examples of the watermarks?

File: blog/2023-08-01-tolgee-with-next-13.md (hunk 2 above)
Comment (JanCizmar) [reply blog/2023-08-01-tolgee-with-next-13.md:11]: If this is a tutorial, no one is interested into information about us facing some challenge. They just want to know how to get started fast.

File: blog/2023-08-01-tolgee-with-next-13.md [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+
+# Setting Up the Configuration
```
Comment (JanCizmar) [inline blog/2023-08-01-tolgee-with-next-13.md:24]: So it's actually a tutorial? I think you should split this to 2 articles. Tutorial with click-bait title, second where you can explain how it works and what was the challenges. That one might be more into how the server components work etc.

File: blog/2023-08-01-tolgee-with-next-13.md [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+
+  return (
+    <TolgeeProvider tolgee={tolgeeSSR} options={{ useSuspense: false }}>
+      {children}
+    </TolgeeProvider>
+  );
+};
+```
+
+The only significant change is the listener for the `permanentChange` event. This event triggers when a translation is updated through an in-context dialog, allowing for a server component refresh with `router.refresh`.
```
Comment (JanCizmar) [inline blog/2023-08-01-tolgee-with-next-13.md:163]: Significant change from what?

File: blog/2023-08-01-tolgee-with-next-13.md [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+
+Now, let's proceed to the server part. As server components don't support React hooks, we need to recreate similar abstractions to those found in `@tolgee/react` ourselves. Fortunately, this process is relatively straightforward; we can essentially utilize vanilla Tolgee, with proper configuration and caching of the instance.
```
Comment (JanCizmar) [inline blog/2023-08-01-tolgee-with-next-13.md:165]: So we want our users to recreate the abstractions? Isn't that why the users use libraries like ours?

File: blog/2023-08-01-tolgee-with-next-13.md [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+
+## Limitations of Server Components
+
+Although in-context translation works with server components, there are some limitations compared to client components. Since the Tolgee cache on the server is separate, Tolgee can't automatically change the translation when creating a screenshot (unlike with client components, which swap the content if you've modified it in a dialog).
+
+Furthermore, if you're using the Tolgee plugin, it won't affect the server's transition to dev mode. As a result, only the client switches, leaving server components non-editable in this mode.
+
+## Conclusion and future steps
+
+Given that server components exist in a completely distinct environment, I might create a separate next.js package in the future. Yet, considering the app router is still in beta and Next.js API adjustments might occur, I opted for this article instead to explain how Tolgee can work with the server.
```
Comment (JanCizmar) [inline blog/2023-08-01-tolgee-with-next-13.md:343]: We are company, so "we might create"

"I opted for this article instead to explain how Tolgee can work with the server." this should be stated at the beginning.

File: blog/2023-08-01-tolgee-with-next-13.md (hunk 1 above)
Comment (stepan662) [reply blog/2023-08-01-tolgee-with-next-13.md:3]: I got like 3 people asking me. Does Tolgee supports next.js router? I think the title is quite alright

**Pull request merged**